ethcore/sync/src/chain/supplier.rs

```diff
@@ -268,6 +268,30 @@ impl SyncSupplier {
         Ok(Some((BlockHeadersPacket, rlp)))
     }
 
+    /// Respond to GetPooledTransactions request
+    fn return_pooled_transactions(io: &dyn SyncIo, r: &Rlp, peer_id: PeerId) -> RlpResponseResult {
+        let payload_soft_limit = io.payload_soft_limit();
+
+        let mut added = 0;
+        let mut rlp = RlpStream::new();
+        rlp.begin_unbounded_list();
+        for v in r {
+            if let Ok(hash) = v.as_val::<H256>() {
+                if let Some(tx) = io.chain().queued_transaction(hash) {
```

Comment:
The transaction might be in a block already, but I suppose it's fine, since the peer should get the same block soon as well and they don't really need the transaction in the pool any more.